src/main/java/be/seeseemelk/mockbukkit/generator/structure/StructureMock.java

```diff
@@ -0,0 +1,36 @@
+package be.seeseemelk.mockbukkit.generator.structure;
+
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonObject;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
+import org.bukkit.generator.structure.Structure;
+import org.bukkit.generator.structure.StructureType;
+import org.jetbrains.annotations.NotNull;
+
+public class StructureMock extends Structure
+{
+
+	private final NamespacedKey key;
+	private final StructureType type;
+
+	public StructureMock(JsonObject data)
+	{
+		this.key = NamespacedKey.fromString(data.get("key").getAsString());
+		this.type = Registry.STRUCTURE_TYPE.get(Preconditions.checkNotNull(NamespacedKey.fromString(data.get("type").getAsString())));
+	}
+
+	@Override
+	public @NotNull StructureType getStructureType()
```

Comment:
Please add a Test for this if possible

Reply:
Added a unit test as of [3cb568f](https://github.com/MockBukkit/MockBukkit/pull/861/commits/3cb568f3f4a495935d48f69d9aad732334dad6fa)